packages/neos-ui-editors/src/Editors/AssetEditor/index.js

```diff
@@ -0,0 +1,109 @@
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import MultiSelectBox from '@neos-project/react-ui-components/src/MultiSelectBox/';
+import AssetOption from '@neos-project/neos-ui-ckeditor-bindings/src/EditorToolbar/AssetOption';
+import {dndTypes} from '@neos-project/neos-ui-constants';
+import {neos} from '@neos-project/neos-ui-decorators';
+import {$get} from 'plow-js';
+
+@neos(globalRegistry => ({
+    assetLookupDataLoader: globalRegistry.get('dataLoaders').get('AssetLookup'),
+    i18nRegistry: globalRegistry.get('i18n')
+}))
+export default class AssetEditor extends PureComponent {
+    state = {
+        options: [],
+        isLoading: false,
+        searchOptions: [],
+        searchTerm: ''
+    };
+
+    static defaultOptions = {
+        // start searching after 2 characters, as it was done in the old UI
+        threshold: 2
+    };
+
+    static propTypes = {
+        value: PropTypes.arrayOf(PropTypes.string),
+        options: PropTypes.object,
+        searchOptions: PropTypes.array,
+        highlight: PropTypes.bool,
+        placeholder: PropTypes.string,
+        displayLoadingIndicator: PropTypes.bool,
+        onSearchTermChange: PropTypes.func,
+        commit: PropTypes.func.isRequired,
+        i18nRegistry: PropTypes.object.isRequired,
+        assetLookupDataLoader: PropTypes.shape({
+            resolveValues: PropTypes.func.isRequired,
+            search: PropTypes.func.isRequired
+        }).isRequired
+    };
+
+    componentDidMount() {
+        this.resolveValue();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.value !== this.props.value) {
+            this.resolveValue();
+        }
+    }
+
+    resolveValue = () => {
+        if (this.props.value) {
+            this.setState({isLoading: true});
+            const resolver = this.props.assetLookupDataLoader.resolveValues.bind(this.props.assetLookupDataLoader);
+            resolver({}, this.props.value)
+                .then(options => {
+                    this.setState({
+                        isLoading: false,
+                        options
+                    });
+                });
+        }
+    }
+
+    handleSearchTermChange = searchTerm => {
+        this.setState({searchTerm});
+        const threshold = $get('options.threshold', this.props) || this.constructor.defaultOptions.threshold;
+        if (searchTerm && searchTerm.length >= threshold) {
+            this.setState({isLoading: true, searchOptions: []});
+            this.props.assetLookupDataLoader.search({}, searchTerm)
+                .then(searchOptions => {
+                    this.setState({
+                        isLoading: false,
+                        searchOptions
+                    });
+                });
+        }
+    }
+
+    handleValueChange = value => {
+        this.setState({searchOptions: [], searchTerm: ''});
+        this.props.commit(value);
+    }
+
+    handleCreateNew = () => {
```

Comment:
do we need this empty method?

Reply:
Currently yes, because it's a required prop, BUT I think that is actually a bug. I need to discuss it with @dimaip when he is in....

Reply:
a right, because the ``MultiSelectBox`` needs the ``onCreateNew`` prop.. hmm... also think we should consider that within the ``MultiSelectBox``. But this is no blocker for now